lualib/resty/kong/upstream.lua

```diff
@@ -0,0 +1,85 @@
+-- Copyright 2019-2022 Kong Inc.
+
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+
+--    http://www.apache.org/licenses/LICENSE-2.0
+
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local _M = {}
+
+local ffi = require("ffi")
+local base = require("resty.core.base")
+base.allows_subsystem("http")
+
+ffi.cdef([[
+int
+ngx_http_lua_ffi_set_next_upstream(ngx_http_request_t *r, uint32_t next_upstream, char **err);
+]])
+
+local type = type
+local C = ffi.C
+local get_request = base.get_request
+local ffi_str = ffi.string
+
+local NGX_OK = ngx.OK
+
+local next_upstream_table = {
+	error = 0x00000002,
+	timeout = 0x00000004,
+	invalid_header = 0x00000008,
+	http_500 = 0x00000010,
+	http_502 = 0x00000020,
+	http_503 = 0x00000040,
+	http_504 = 0x00000080,
+	http_403 = 0x00000100,
+	http_404 = 0x00000200,
+	http_429 = 0x00000400,
+	off = 0x00001000,
+	non_idempotent = 0x00004000,
+}
+
+function _M.set_next_upstream(...)
+	local nargs = select("#", ...)
+	if nargs == 0 then
+		return "no argument"
```

Comment:
indentation

Reply:
indentation is 4 spaces, what's the problem here?